event_sourcery/event_store.py

```diff
@@ -160,3 +160,10 @@ def _serialize_events(
             )
             for event in events
         ]
+
+    def iter(self, batch_size: int = 100) -> Iterator[Metadata]:
```

Comment:
@lzukowski (not calling the other Luke cause he has busy weekend) - I was thinking there are two cases.
1) in one, we'd like to have finite generator that will return events in batches until it catches up with whatever is in the store

2) in the second one, we'd like to iter indefinitely - including waiting & polling for new events.

Do you have any ideas regarding the API? I could implement `Iterator` design pattern as such, but I was also thinking about having just two generator methods - `iter` and `iter_infinite` or something similar.

WDYT?